### DISCUSSION TITLE
Feature Request: hot switching in/out Diag ROMs?

### DISCUSSION THREAD

**Discussion by turnkit** [misterblack1/trs80-diagnosticrom#19] [Ideas]
Would it be possible to create a daughter board that would take both the diagnostic ROM and the original boot ROM?

What I would like to see is a toggle (or push-button) switch on the board for swapping ROMs, and an option in the diagnostic ROM to move the ROM code into RAM, preparing for a switch back to the original ROM. 

Additionally, if possible, the ROM bank switching could also occur via a "soft switch" e.g. on the Model 3 via an OUT 242 [or other not typically used port], 1 [for ROM socket 1].

For example the on-screen option might be: Move diagnostic program to: F000.  
Use this option prior to switching back to original ROMs in order to run ROM checksum diagnostic.

And then, assuming this can be done without glitching the system, an extra option can appear on the diagnostic screen to do a ROM checksum.  Ideally the program can check for all the ROM checksums listed here:

- https://www.trs-80.com/wordpress/roms/checksums-mod-1/
- https://www.trs-80.com/wordpress/roms/checksums-mod-3/
- https://www.trs-80.com/wordpress/roms/checksums-mod-4/
- https://www.trs-80.com/wordpress/roms/checksums-clones/
 
 (And for any indicating "MISSING!" please add a note -- "This ROM has not been archived please email the site admin at trs-80.org" so that we can get all the missing ROMs available for emulators.)

**Comment by kiwisincebirth** [comment]
Here is one solution

https://github.com/RetroStack/2332_2364-ROM_Adapter
